src/iris/ui/static/js/iris.js

```diff
@@ -1972,6 +1994,20 @@ iris = {
         iris.createAlert('Error: Failed to load reprioritization data -' + response.text);
       });
     },
+    getCustomNotificationSettings: function() {
+      var self = this;
+      return $.getJSON(this.data.url + this.data.user + '/categories').done(function(response){
+        self.data.settings.categories = response;
+        var tempSet = new Set();
+        self.data.settings.categories.forEach(function (item, index) {
+          tempSet.add(item['application']);
+        });
+        self.data.settings['customApp'] = Array.from(tempSet);
```

Comment:
As we're already using sets, why not just set self.data.settings.customApp to the set variable itself, instead of making it an array? 

That way you could use the native methods Set objects have (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set ) , instead of using splice/indexOf to check for presence/remove items.

Reply:
changed it to use set

Reply:
changed to using sets